runtime/protocol/http-client-engines/test-suite/jvm/src/aws/smithy/kotlin/runtime/http/test/util/TestServer.kt

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package aws.smithy.kotlin.runtime.http.test.util
+
+import aws.smithy.kotlin.runtime.http.test.suite.concurrentTests
+import aws.smithy.kotlin.runtime.http.test.suite.downloadTests
+import aws.smithy.kotlin.runtime.http.test.suite.uploadTests
+import io.ktor.application.*
+import io.ktor.request.*
+import io.ktor.response.*
+import io.ktor.routing.*
+import io.ktor.server.cio.*
+import io.ktor.server.engine.*
+import redirectTests
+import java.io.Closeable
+import java.util.concurrent.TimeUnit
+
+private const val DEFAULT_PORT = 8082
+
+private class Resources : Closeable {
+    private val resources = mutableListOf<Closeable>()
+
+    fun add(resource: Closeable) {
+        resources.add(resource)
+    }
+
+    override fun close() {
+        resources.forEach(Closeable::close)
+    }
+}
```

Comment:
**Question**: Why is a list of resources necessary? Seems like we'll only ever have one.